scripts/build/termux_download_deb_pac.sh

```diff
@@ -1,126 +1,79 @@
 #!/usr/bin/bash
 
 termux_download_deb_pac() {
-	local PACKAGE=$1
-	local PACKAGE_ARCH=$2
-	local VERSION=$3
-	local VERSION_PACMAN=$4
-
-	local PKG_FILE
-	if [ "$TERMUX_REPO_PKG_FORMAT" = "debian" ]; then
-		PKG_FILE="${PACKAGE}_${VERSION}_${PACKAGE_ARCH}.deb"
-	elif [ "$TERMUX_REPO_PKG_FORMAT" = "pacman" ]; then
-		PKG_FILE="${PACKAGE}-${VERSION_PACMAN}-${PACKAGE_ARCH}.pkg.tar.xz"
+	local PACKAGE=$1 PACKAGE_ARCH=$2 VERSION=$3 VERSION_PACMAN=$4
+	local PKG_FILE= PKG_HASH="" debian=false pacman=false without_depversion_binding=false
+	case "$TERMUX_REPO_PKG_FORMAT" in
+		debian) PKG_FILE="${PACKAGE}_${VERSION}_${PACKAGE_ARCH}.deb"; debian=true;;
+		pacman) PKG_FILE="${PACKAGE}-${VERSION_PACMAN}-${PACKAGE_ARCH}.pkg.tar.xz"; pacman=true;;
+	esac
+	if [[ -n "$TERMUX_WITHOUT_DEPVERSION_BINDING" && "$TERMUX_WITHOUT_DEPVERSION_BINDING" == true ]]; then
+		without_depversion_binding=true
 	fi
-	PKG_HASH=""
 
 	# Dependencies should be used from repo only if they are built for
 	# same package name.
 	# The data.tar.xz extraction by termux_step_get_dependencies would
 	# extract files to different prefix than TERMUX_PREFIX and builds
 	# would fail when looking for -I$TERMUX_PREFIX/include files.
-	if [ "$TERMUX_REPO_APP__PACKAGE_NAME" != "$TERMUX_APP_PACKAGE" ]; then
+	if [[ "$TERMUX_REPO_APP__PACKAGE_NAME" != "$TERMUX_APP_PACKAGE" ]]; then
 		echo "Ignoring download of $PKG_FILE since repo package name ($TERMUX_REPO_APP__PACKAGE_NAME) does not equal app package name ($TERMUX_APP_PACKAGE)"
 		return 1
 	fi
 
-	if [ "$TERMUX_ON_DEVICE_BUILD" = "true" ]; then
+	if [[ "$TERMUX_ON_DEVICE_BUILD" = true ]]; then
 		case "$TERMUX_APP_PACKAGE_MANAGER" in
-			"apt") apt install -y "${PACKAGE}$(test ${TERMUX_WITHOUT_DEPVERSION_BINDING} != true && echo "=${VERSION}")";;
-			"pacman") pacman -S "${PACKAGE}$(test ${TERMUX_WITHOUT_DEPVERSION_BINDING} != true && echo "=${VERSION_PACMAN}")" --needed --noconfirm;;
+			"apt") apt install -y "$PACKAGE$(! $without_depversion_binding && echo "=$VERSION" || :)";;
```

Comment:
You can just not use the `!` and invert the ternary operators.
```suggestion
			"apt") apt install -y "$PACKAGE$($without_depversion_binding || echo "=$VERSION")";;
```
Or we should probably just not invoke `true` as a program here.
```suggestion
			"apt") apt install -y "$PACKAGE$([[ "$without_depversion_binding" == 'true' ]] && echo "=$VERSION")";;
```
Same goes for `pacman`.